server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/VacationRoutesTest.java

```diff
@@ -0,0 +1,297 @@
+        assertThat(vacation.getSubject()).isEqualTo(Optional.of("On vacation again"));
+        assertThat(vacation.getTextBody()).isEqualTo(Optional.of("Need more vacation!"));
+        assertThat(vacation.getHtmlBody()).isEqualTo(Optional.of("<p>Need more vacation!</p>"));
+
+        Boolean registered = notificationRegistry.isRegistered(bob, alice).block();
+        assertThat(registered).isFalse();
+    }
+
+    @Test
+    void patchVacationModifies() throws Exception {
+        AccountId bob = AccountId.fromString(BOB);
+        RecipientId alice = RecipientId.fromMailAddress(new MailAddress(ALICE));
+
+        vacationRepository.modifyVacation(bob, VacationPatch.builderFrom(VACATION).build())
+            .then(notificationRegistry.register(bob, alice, Optional.empty())).block();
+
+        given()
+            .body("{\"enabled\":true,\"subject\":\"More vacation\"}")
+        .when()
+            .patch(VacationRoutes.VACATION + SEPARATOR + BOB)
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204);
+
+        Vacation vacation = vacationRepository.retrieveVacation(bob).block();
+        assertThat(vacation).isNotNull();
+        assertThat(vacation.isEnabled()).isTrue();
+        assertThat(vacation.getFromDate()).isEmpty();
+        assertThat(vacation.getToDate()).isEmpty();
+        assertThat(vacation.getSubject()).isEqualTo(Optional.of("More vacation"));
+        assertThat(vacation.getTextBody()).isEmpty();
+        assertThat(vacation.getHtmlBody()).isEmpty();
+
+        Boolean registered = notificationRegistry.isRegistered(bob, alice).block();
+        assertThat(registered).isFalse();
+    }
+
+    @Test
+    void patchVacationFails() {
+        given()
+            .body("{\"enabled\":true,\"subject\":\"On vacation again\"}")
+        .when()
+            .patch(VacationRoutes.VACATION + SEPARATOR + CAROL)
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400);
+    }
+
+    @Test
+    void deleteVacation() throws Exception {
+        AccountId bob = AccountId.fromString(BOB);
+        RecipientId alice = RecipientId.fromMailAddress(new MailAddress(ALICE));
+
+        vacationRepository.modifyVacation(bob, VacationPatch.builderFrom(VACATION).build())
+            .then(notificationRegistry.register(bob, alice, Optional.empty())).block();
+
+        when()
+            .delete(VacationRoutes.VACATION + SEPARATOR + BOB)
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204);
+
+        Vacation vacation = vacationRepository.retrieveVacation(bob).block();
+        assertThat(vacation).isNotNull();
+        assertThat(vacation.isEnabled()).isFalse();
+        assertThat(vacation.getFromDate()).isEmpty();
+        assertThat(vacation.getToDate()).isEmpty();
+        assertThat(vacation.getSubject()).isEmpty();
+        assertThat(vacation.getTextBody()).isEmpty();
+        assertThat(vacation.getHtmlBody()).isEmpty();
```

Comment:
When we have many assertions we generally wrap them in a soft assertion:

```
SoftAssertions.assertThatSoftly(softly -> {
    softly.assertThat(vacation).isNotNull();
    softly.assertThat(vacation.isEnabled()).isFalse();
    // ...
});
```

In case of failures it executes all the statements and do a global reports instead of failing on the first one.

This enables fixing all the issues at once.

Reply:
Sure, I will change it accordingly. And good to know, I sow this elsewhere and was wondering what it was for.